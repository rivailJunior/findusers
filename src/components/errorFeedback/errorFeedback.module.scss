@use '../../pages/index.scss' as index;

@mixin errorLabel($fontSize: 16) {
    @include index.default-label(14px, 1%, 100%, 500);
    font-size: $fontSize;
}

.errorContainer > div {
    @include errorLabel(20px);
    display: flex;
    flex-direction: column;
    width: 100%;
    min-height: 50px;
    color: white;
    background-color: lightcoral;
    justify-content: center;
    border-radius: 2px;
}

.subTitle {
    @include errorLabel(14);
}
