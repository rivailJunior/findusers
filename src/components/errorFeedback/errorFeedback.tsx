import React from 'react';
import styles from './errorFeedback.module.scss';

type iErrorFeedback = {
    title: string
    subTitle?: string
}
const ErrorFeedback = ({ title, subTitle }: iErrorFeedback) => {
    return (
        <div className={styles.errorContainer}>
            <div>{title}</div>
            {subTitle && <div data-testid="subTitleError" className={styles.subTitle}>{subTitle}</div>}
        </div>
    );
}

export { ErrorFeedback };
