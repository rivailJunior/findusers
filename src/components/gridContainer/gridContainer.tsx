import React, { Children } from 'react';
import styles from './container.module.scss'
type iContainer = {
    children?: JSX.Element[] | JSX.Element
}
const GridContainer = ({ children }: iContainer): JSX.Element => {
    return (
        <div data-testid="gridContainer" className={styles.mainContainer}>
            {Children.map(children, (child) => {
                return child
            })}
        </div>
    );
}

type iItem = {
    size: number,
    children?: JSX.Element[] | JSX.Element
}
const GridContainerItem = ({ size, children }: iItem): JSX.Element => {
    const itemSize = `containerItem${ size }`
    return (
        <div data-testid="gridContainerItem" className={styles[itemSize]}>
            {Children.map(children, (child) => {
                return child
            })
            }</div>
    )
}

export { GridContainer, GridContainerItem };
