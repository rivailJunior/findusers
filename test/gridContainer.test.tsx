import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { ErrorFeedback } from '../src/components/errorFeedback/errorFeedback'
import { GridContainer, GridContainerItem } from '../src/components/gridContainer/gridContainer'

afterEach(cleanup)

describe('Grid Contaiiner', () => {
    test('Render Correctly', () => {
        const container = renderer.create(<GridContainer >
            <ErrorFeedback title="Nenhum repositorio encontrado" />
        </GridContainer>)
        container.toJSON();
        expect(container).toMatchSnapshot()
    })

    test('Wrapper a child component', () => {
        const { getByText } = render(<GridContainer >
            <ErrorFeedback title="Nenhum repositorio encontrado!" />
        </GridContainer>)
        expect(getByText(/Nenhum repositorio encontrado!/i)).toBeInTheDocument();
    })
    test('Wrapper more than one child component', () => {
        const { getByText, getByTestId } = render(<GridContainer >
            <ErrorFeedback title="Nenhum repositorio encontrado!" />
            <ErrorFeedback title="Parece que deu bom!" />
        </GridContainer>)
        expect(getByText(/Nenhum repositorio encontrado!/i)).toBeInTheDocument();
        expect(getByTestId('gridContainer').childNodes).toHaveLength(2)
    })

    test('Render grid without any child componenent', () => {
        const { getByText, getByTestId } = render(<GridContainer />)
        expect(getByTestId('gridContainer').childNodes).toHaveLength(0);
    })

    test('Render grid item with correct size class ', () => {
        const { getByTestId } = render(<GridContainer >
            <GridContainerItem size={12} >
                <ErrorFeedback title="Nenhum item encontrado" />
            </GridContainerItem>
        </GridContainer>)
        expect(getByTestId('gridContainerItem')).toHaveClass('containerItem12');
    })

    test('Render grid item without any child componenent', () => {
        const { getByTestId } = render(<GridContainer >
            <GridContainerItem size={12} />
        </GridContainer>)
        expect(getByTestId('gridContainerItem').childNodes).toHaveLength(0);
    })
})