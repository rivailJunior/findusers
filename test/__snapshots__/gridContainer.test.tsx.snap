// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Contaiiner Render Correctly 1`] = `
<div
  className="mainContainer"
  data-testid="gridContainer"
>
  <div
    className="errorContainer"
  >
    <div>
      Nenhum repositorio encontrado
    </div>
  </div>
</div>
`;
